portable/GCC/ARM_CM4F/port.c

```diff
@@ -656,13 +683,24 @@ void xPortSysTickHandler( void )
 				portNVIC_SYSTICK_LOAD_REG = ( ( ulCompleteTickPeriods + 1UL ) * ulTimerCountsForOneTick ) - ulCompletedSysTickDecrements;
 			}
 
-			/* Restart SysTick so it runs from portNVIC_SYSTICK_LOAD_REG
-			again, then set portNVIC_SYSTICK_LOAD_REG back to its standard
-			value. */
+			/* Restart SysTick so it runs from portNVIC_SYSTICK_LOAD_REG again,
+			then set portNVIC_SYSTICK_LOAD_REG back to its standard value.  If
+			the SysTick is not using the core clock, temporarily configure it to
+			use the core clock.  This configuration forces the SysTick to load
+			from portNVIC_SYSTICK_LOAD_REG immediately instead of at the next
+			cycle of the other clock.  Then portNVIC_SYSTICK_LOAD_REG is ready
+			to receive the standard value immediately. */
 			portNVIC_SYSTICK_CURRENT_VALUE_REG = 0UL;
-			portNVIC_SYSTICK_CTRL_REG |= portNVIC_SYSTICK_ENABLE_BIT;
-			vTaskStepTick( ulCompleteTickPeriods );
+			portNVIC_SYSTICK_CTRL_REG = portNVIC_SYSTICK_CLK_BIT | portNVIC_SYSTICK_INT_BIT | portNVIC_SYSTICK_ENABLE_BIT;
 			portNVIC_SYSTICK_LOAD_REG = ulTimerCountsForOneTick - 1UL;
+			#if( portNVIC_SYSTICK_CLK_BIT_SETTING != portNVIC_SYSTICK_CLK_BIT )

```

Comment:
May need some explanation on this #if-#endif. 

My understanding of this #if-#endif --
if using external clock (bit-2 is 0), it seems like we are trying to re-enable SysTick. 
Would it be necessary?

Reply:
The new code comments attempt to explain.  This #if - #endif block changes the clocking configuration of the systick from using the core clock to using the alternate clock but only in configurations that use the alternate clock.  We detect configurations that use the alternate clock by checking portNVIC_SYSTICK_CLK_BIT_SETTING.